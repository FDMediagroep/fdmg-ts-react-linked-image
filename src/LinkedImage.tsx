import * as React from 'react';
import A from 'fdmg-ts-react-anchor';
import Image from 'fdmg-ts-react-image';
import {MouseEvent} from 'react';

export interface Props {
    alt: string;
    ariaLabel?: string;
    className?: string;
    href: string;
    onClick?: (event: MouseEvent<HTMLAnchorElement>) => void;
    src: string;
    target?: string;
}

/**
 * Renders an image wrapped in an anchor.
 */
export default class LinkedImage extends React.Component<Props, any> {
    public state: any;
    public props: Props;

    constructor(props: Props) {
        super(props);
        this.props = props;
        console.info('LinkedImage composition', props);
    }

    render() {
        return (
            <A href={this.props.href}
                target={this.props.target}
                className={this.props.className}
                aria-label={this.props.ariaLabel?this.props.ariaLabel:this.props.alt}
                onClick={this.props.onClick}>
                <Image src={this.props.src} alt={this.props.alt}/>
            </A>
        );
    }
}
