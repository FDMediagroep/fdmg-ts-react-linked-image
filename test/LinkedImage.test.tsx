import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { mount } from 'enzyme';

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import LinkedImage from "../src/LinkedImage";

console.info = function() {};

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

test('LinkedImage renders correctly', () => {
    let component = renderer.create(
        <LinkedImage href={'url-link'} alt={'alt-text'} className={'css-class-name'} src={'image-src'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <LinkedImage href={'url-link'} target={'new_window'} alt={'alt-text'} className={'css-class-name'} src={'image-src'}/>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <LinkedImage href={'url-link'} target={'new_window'} ariaLabel={'aria-label'} alt={'alt-text'} className={'css-class-name'} src={'image-src'}/>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('LinkedImage onClick callback handling', () => {
    const spy = jest.fn();
    let component = mount(
        <LinkedImage onClick={spy} alt={"alt-text"} src={'test.png'} className={'css-class-name'}/>
    );

    component.simulate('click');
    expect(spy).toHaveBeenCalled();
});
