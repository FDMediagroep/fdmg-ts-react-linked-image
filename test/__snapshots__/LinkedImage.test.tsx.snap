// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedImage renders correctly 1`] = `
<a
  aria-label={undefined}
  className="css-class-name"
  href="url-link"
  onClick={undefined}
  tabIndex={0}
  target={undefined}
  title={undefined}
>
  <img
    alt="alt-text"
    className={undefined}
    src="image-src"
  />
</a>
`;

exports[`LinkedImage renders correctly 2`] = `
<a
  aria-label={undefined}
  className="css-class-name"
  href="url-link"
  onClick={undefined}
  tabIndex={0}
  target="new_window"
  title={undefined}
>
  <img
    alt="alt-text"
    className={undefined}
    src="image-src"
  />
</a>
`;

exports[`LinkedImage renders correctly 3`] = `
<a
  aria-label={undefined}
  className="css-class-name"
  href="url-link"
  onClick={undefined}
  tabIndex={0}
  target="new_window"
  title={undefined}
>
  <img
    alt="alt-text"
    className={undefined}
    src="image-src"
  />
</a>
`;
